utils/make_changelog.py

```diff
@@ -0,0 +1,385 @@
+            logging.warning('Not found author for commit {}.'.format(commit_info['html_url']))
+
+    return users
+
+
+# List of unknown commits -> text description.
+def process_unknown_commits(commits, commits_info, users):
+
+    pattern = 'Commit: [{}]({})\nAuthor: {}\nMessage: {}'
+
+    texts = []
+
+    for commit in commits:
+        info = commits_info[commit]
+        html_url = info['html_url']
+        msg = info['commit']['message']
+
+        # GitHub login
+        login = info['author']['login']
+        name = None
+
+        # Firstly, try get name from github user
+        try:
+            name = users[login]['name']
+        except:
+            pass
+
+        # Then, try get name from commit
+        if not name:
+            try:
+                name = info['commit']['author']['name']
+            except:
+                pass
+
+        author = '[{}]({})'.format(name or login, info['author']['html_url'])
+        texts.append(pattern.format(commit, html_url, author, msg))
+
+    text = 'Commits which are not from any pull request:\n\n'
+    return text + '\n\n'.join(texts)
+
+
+# List of pull requests -> text description.
+def process_pull_requests(pull_requests, users, repo):
+    groups = {}
+
+    for id, item in pull_requests.items():
+        lines = list(filter(len, map(lambda x: x.strip(), item['description'].split('\n'))))
+
+        cat_pos = None
+        short_descr_pos = None
+        long_descr_pos = None
+
+        if lines:
+            for i in range(len(lines) - 1):
+                if re.match('^\**Category\s*(\(leave one\))*:*\**\s*$', lines[i]):
+                    cat_pos = i
+                if re.match('^\**\s*Short description', lines[i]):
+                    short_descr_pos = i
+                if re.match('^\**\s*Detailed description', lines[i]):
+                    long_descr_pos = i
+
+        cat = ''
+        if cat_pos:
+            # TODO: Sometimes have more than one
+            cat = lines[cat_pos + 1]
+        cat = cat.strip().lstrip('-').strip()
+
+        short_descr = ''
+        if short_descr_pos:
+            short_descr_end = long_descr_pos or len(lines)
+            short_descr = lines[short_descr_pos + 1]
+            if short_descr_pos + 2 != short_descr_end:
+                short_descr += ' ...'
+
+        # TODO: Add detailed description somewhere
+
+        pattern = u"{} [#{}]({}) ({})"
+        link = 'https://github.com/{}/pull/{}'.format(repo, id)
+        author = 'author not found'
+        if item['user'] in users:
+            # TODO get user from any commit if no user name on github
+            user = users[item['user']]
+            author = u'[{}]({})'.format(user['name'] or user['login'], user['html_url'])
+
+        if cat not in groups:
+            groups[cat] = []
+        groups[cat].append(pattern.format(short_descr, id, link, author))
+
+    texts = []
+    for group, text in groups.items():
+        items = [u'* {}'.format(pr) for pr in text]
```

Comment:
Seems like `u'str'` doesn't make sense in python3 https://www.python.org/dev/peps/pep-0414/.

Reply:
But you want to support python 2, than OK.

Reply:
Yes, now it works for both pythons for me.